json_object.c

```diff
@@ -618,7 +623,10 @@ static int json_object_int_to_json_string(struct json_object* jso,
 {
 	/* room for 19 digits, the sign char, and a null term */
 	char sbuf[21];
-	snprintf(sbuf, sizeof(sbuf), "%" PRId64, jso->o.c_int64);
+	if (jso->o.c_int.cint_type == json_object_int_type_int64)
+	  snprintf(sbuf, sizeof(sbuf), "%" PRId64, jso->o.c_int.cint.c_int64);
+	else if (jso->o.c_int.cint_type == json_object_int_type_uint64)
```

Comment:
Let make this just a plain "else", to ensure we write *something* to sbuf.  Please adjust the other cint_type if conditions and switch statements similarly, to ensure we have a more definite action taken rather than just falling through.

Reply:
ok, done it